echo_test_server.py

```diff
@@ -0,0 +1,105 @@
+import asyncio
+import os
+import signal
+import sys
+from textwrap import dedent
+
+try:
+    from rich import print
+except ImportError:
+    pass
+
+os.environ["RUST_LOG"] = "mitmproxy_wireguard=debug"
+
+import mitmproxy_wireguard
+
+# (private key, public key)
+server_keypair = ("EG47ZWjYjr+Y97TQ1A7sVl7Xn3mMWDnvjU/VxU769ls=", "mitmV5Wo7pRJrHNAKhZEI0nzqqeO8u4fXG+zUbZEXA0=")
+client_keypair = ("qG8b7LI/s+ezngWpXqj5A7Nj988hbGL+eQ8ePki0iHk=", "Test1sbpTFmJULgSlJ5hJ1RdzsXWrl3Mg7k9UTN//jE=")
+
+
+async def main():
+    print(f"{dir(mitmproxy_wireguard)=}")
+
+    k = mitmproxy_wireguard.genkey()
+    print(f"mitmproxy_wireguard.genkey()={k!r}")
+    print(f"{mitmproxy_wireguard.pubkey(k)=}")
+
+    def receive_datagram(data, src_addr, dst_addr):
+        print(f"Received datagram: {data=} {src_addr=} {dst_addr=}")
+        server.send_datagram(data.upper(), dst_addr, src_addr)
+        print("Echoed datagram.")
+
+    server = await mitmproxy_wireguard.start_server(
+        "0.0.0.0",
+        51820,
+        server_keypair[0],
+        [client_keypair[1]],
+        handle_connection,
+        receive_datagram
+    )
```

Comment:
This is the main Python API now - start a server and specify a callback for datagrams and TCP connections.